modules/user/user.repository.js

```diff
@@ -1,39 +1,60 @@
-const fsPromises = require("fs").promises;
-const filePath = "./database/users.json";
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
 const { CustomError } = require("../../errorHandler");
-const { isArrayWithData } = require("../../utils");
+require("dotenv").config();
+const url = process.env.MONGODB_URL;
+
+const userScheme = new Schema({
+    id: Number,
+    name: String,
+    surname: String,
+    email: String,
+    phone: Number,
+});
+const UserModel = mongoose.model("User", userScheme);
+
+const userAggregateForm = {
+    _id: 0,
+    id: 1,
+    name: 1,
+    surname: 1,
+    email: 1,
+    phone: 1,
+};
+
+mongoose.connect(url, {
+    useUnifiedTopology: true,
+    useNewUrlParser: true,
+});
 
 class UserRepository {
     async readAllUsers() {
-        const data = await fsPromises.readFile(filePath);
-        const usersArr = await JSON.parse(data);
+        const usersArr = await UserModel.find({}, userAggregateForm);
 
-        if (isArrayWithData(usersArr)) {
-            return usersArr;
-        } else {
-            throw new Error(500);
-        }
+        return usersArr;
     }
 
-    async findUserIndexById(id) {
-        const usersArr = await this.readAllUsers();
-        let userIndex;
+    async findMongoIdByUserId(userId) {
+        const user = await UserModel.findOne({ id: userId }, { _id: 1 });
 
-        for (let i = 0; i < usersArr.length; i++) {
-            if (usersArr[i].id == id) {
-                userIndex = i;
-                return userIndex;
-            }
+        if (!user) {
+            throw new CustomError("CANT_FIND_USER_BY_ID");
         }
 
-        throw new CustomError("CANT_FIND_USER_BY_ID");
+        const mongoId = user._id;
+
+        return mongoId;
     }
 
-    async readUserById(id) {
-        const usersArr = await this.readAllUsers();
-        const index = await this.findUserIndexById(id);
+    async readUserById(userId) {
+        const mongoId = await this.findMongoIdByUserId(userId);
```

Comment:
that looks overhead to have two queries to DB here. Let's try get user with one query

- [ ] get user by id with one query